src/facilities/costCenters.js

```diff
@@ -0,0 +1,300 @@
+      .get(`${this._baseUrl}/costcenters`)
+      .then((costCenters) => costCenters.map(formatCostCenterFromServer));
+  }
+
+  /**
+   * Get a listing of all cost centers for an organization
+   *
+   * API Endpoint: '/organizations/:organizationId/costcenters'
+   * METHOD: GET
+   *
+   * @param {string} organizationId The ID of the organization (formatted as a UUID)
+   *
+   * @returns {Promise}
+   * @fulfill {CostCenter[]}
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.facilities.costCenters
+   *   .getAllByOrganizationId('59270c25-4de9-4b22-8e0b-ab287ac344ce')
+   *   .then((costCenters) => console.log(costCenters))
+   *   .catch((err) => console.log(err));
+   */
+  getAllByOrganizationId(organizationId) {
+    if (!organizationId) {
+      return Promise.reject(
+        new Error(
+          "An organization id is required for getting a list of an organization's cost centers."
+        )
+      );
+    }
+
+    return this._request
+      .get(`${this._baseUrl}/organizations/${organizationId}/costcenters`)
+      .then((costCenters) => costCenters.map(formatCostCenterFromServer));
+  }
+
+  /**
+   * Removes a facility from a cost center
+   *
+   * API Endpoint: '/costcenters/:costCenterId/facility/:facilityId'
+   * Method: DELETE
+   *
+   * @param {string} costCenterId UUID corresponding with a cost center
+   * @param {number} facilityId ID corresponding with the facility
+   *
+   * @returns {Promise}
+   * @fulfill {undefined}
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.facilities.costCenters.removeFacility('b3dbaae3-25dd-475b-80dc-66296630a8d0', 4)
+   *   .catch((err) => console.log(err));
+   */
+  removeFacility(costCenterId, facilityId) {
+    let errorMsg;
+
+    if (!costCenterId) {
+      errorMsg =
+        'A costCenterId is required to remove a relationship between a cost center and a facility.';
+    } else if (!facilityId) {
+      errorMsg =
+        'A facilityId is required to remove a relationship between a cost center and a facility.';
+    }
+
+    if (errorMsg) {
+      return Promise.reject(new Error(errorMsg));
+    }
+
+    return this._request.delete(
+      `${this._baseUrl}/costcenters/${costCenterId}/facility/${facilityId}`
+    );
+  }
+
+  /**
+   * Updates an existing cost center
+   *
+   * API Endpoint: '/costcenters/:costCenterId'
+   * Method: PUT
+   *
+   * @param {String} costCenterId
+   * @param {Object} update
+   * @param {string} [update.description]
+   * @param {string} [update.name]
+   *
+   * @returns {Promise}
+   * @fulfill {FacilityGrouping} Information about the updated cost center
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.facilities.costCenters
+   *   .update({
+   *     description: 'US States of CT, MA, ME, NH, RI, VT',
+   *     name: 'New England, USA',
+   *   })
+   *   .then((costCenter) => console.log(costCenter))
+   *   .catch((err) => console.log(err));
+   */
+  update(costCenterId, update) {
+    if (!costCenterId) {
```

Comment:
Minor, [up top](https://github.com/ndustrialio/contxt-sdk-js/pull/34/files#diff-46620d5ea7e11fbc46891e5d4413fa20R228) we are creating a variable for the errors and rejecting in one spot, here we are rejecting on each conditional.

Reply:
On this one, we are checking two things on the `update` with different error messages. If the first is not true, the second is also not true, but overwriting the error message.